src/lib/eslint/BaseEslintEngine.ts

```diff
@@ -199,6 +197,22 @@ export abstract class BaseEslintEngine implements RuleEngine {
 				// get run-config for the engine and add to config
 				Object.assign(config, await this.strategy.getRunConfig(engineOptions));
 
+				// TODO: This whole code block is part of a fix to W-7791882, which was known from the start to be sub-optimal.
+				//       It requires too much leaking of the abstraction. So during the 3.0 cycle, we should replace it with
+				//       something better.
+				// From https://eslint.org/docs/developer-guide/nodejs-api:
+				// options.baseConfig. Configuration object, extended by all configurations used with this instance.
+				// You can use this option to define the default settings that will be used if your configuration files don't configure it.
+				// If they don't already have a baseConfig property, we'll need to instantiate one.
+				config['baseConfig'] = config['baseConfig'] || {'env': {}};
```

Comment:
A couple of stylistic nitpicks:
1. Can we declare constants (module level should be good enough) for "baseConfig" and "env" so we don't have accidental typos while making changes to this code another time?
2. How about extracting the new portion into a method of its own? That way code readability will be focused.